api/tests/test_asset_api.py

```diff
@@ -188,6 +188,47 @@ def test_admin_can_filter_asset_by_asset_status(self, mock_verify_id_token):
         )
         self.assertEqual(response.data["count"], count + 1)
 
+    @patch("api.authentication.auth.verify_id_token")
+    def test_admin_can_filter_asset_by_asset_department(self, mock_verify_id_token):
+        self.asset_3.assigned_to = self.asset_assignee_department
+        self.asset_3.save()
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        response = client.get(
+            "{}?department={}".format(
+                self.manage_asset_urls, self.department_travel.id
+            ),
+            HTTP_AUTHORIZATION="Token {}".format(self.token_admin),
+        )
+
+        self.assertEqual(response.data["results"][0]["id"], self.asset_3.id)
+        self.assertEqual(
+            response.data["results"][0]["asset_category"], self.asset_3.asset_category
+        )
+        self.assertEqual(
+            response.data["results"][0]["asset_sub_category"],
+            self.asset_3.asset_sub_category,
+        )
+        self.assertEqual(
+            response.data["results"][0]["asset_make"], self.asset_3.asset_make
+        )
+        self.assertEqual(
+            response.data["results"][0]["asset_code"], self.asset_3.asset_code
+        )
+        self.assertEqual(
+            response.data["results"][0]["asset_type"], self.asset_3.asset_type
+        )
+        self.assertEqual(
+            response.data["results"][0]["serial_number"], self.asset_3.serial_number
+        )
+        self.assertEqual(
+            response.data["results"][0]["asset_location"],
+            self.asset_3.asset_location.name,
+        )
+        self.assertEqual(
+            response.data["results"][0]["assigned_to"]["id"],
+            self.asset_assignee_department.department_id,
+        )
```

Comment:
Pretty comprehensive assertions. It's also important to assert the number of asset expected after filtering.